Build and update kernel variants through linux-pkg

sdimitro opened the pull request:
Build and update kernel variants through linux-pkg

    = Motivation

    The Ubuntu kernels have served us well but we currently have two
    problems with them:

    1] When we trigger a kernel bug we have to wait until a patch is
       released by Canonical. This may sound like a rare event but
       we've already hit bugs in iSCSI and NFS subsystems that we've
       had to work around.

    2] There are certain kernel config options that we'd like to enable
       that won't be enabled by Canonical until later releases or at
       all even. Examples include BTF info in vmlinux and the PSI api
       needed by oomd.

    = This Patch

    Enable linux-pkg to compile the Ubuntu kernels from scratch. In
    addition it allows us to base our own patches on top of these
    kernels, and rebase them whenever we get new commits from the
    upstream Canonical kernels. This allows us to:

    1] Fix upstream kernel bugs in a timely manner during customer and
       internal escalations.

    2] Customize our kernel config as we see fit.

    3] Have at least one standard workflow for experimenting with
       our own kernel changes that we may want to potentially upstream.

sdimitro commented:
cc @pzakha - not expecting a thorough review. This is mostly to demonstrate progress

pzakha commented on default-package-config.sh:
```diff
@@ -34,3 +34,107 @@ function store_build_info() {
+	# Clean up everything generated so far and recreate the
+	# final control file with the arguments that we want.
+	#
+	logmust fakeroot debian/rules clean
+	logmust debian/rules debian/control ${debian_rules_args}
```
Could you just run ```logmust fakeroot debian/rules clean ${debian_rules_args}```?
`clean` ends up running `debian/control`

sdimitro replied:
👍

sdimitro commented:
@pzakha updated the PR with the second milestone

pzakha commented on default-package-config.sh:
```diff
@@ -34,3 +34,268 @@ function store_build_info() {
+	local dlpx_patch_end dlpx_patch_start current_ubuntu_commit
+	dlpx_patch_start=$(git log --pretty=oneline repo-HEAD | grep @@DELPHIX_PATCHSET_START@@ | awk '{ print $1 }')
+	[ -z "${dlpx_patch_start}" ] && die "could not find DELPHIX_PATCHSET_START"
+	current_ubuntu_commit=$(git log --pretty=oneline repo-HEAD | grep -A 1 @@DELPHIX_PATCHSET_START@@ | tail -n 1 | awk '{ print $1 }')
```
nit: You can also use `current_ubuntu_commit=$(git rev-parse ${dlpx_patch_start}^)`

pzakha commented on default-package-config.sh:
```diff
@@ -34,3 +34,268 @@ function store_build_info() {
+	#
+	local upstream_tag_info
+	upstream_tag_info=$(git ls-remote --tags --ref upstream | grep "${tag_prefix}" | tail -n 1)
+	[ -z "${upstream_tag_info}" ] && die "could not find upstream tag for tag prefix: ${tag_prefix}"
```
nit: for consistency, it would be preferable to use `[[` everywhere

pzakha commented on packages/linux-kernel-aws/config.sh:
```diff
@@ -0,0 +1,35 @@
+
+# shellcheck disable=SC2034
+DEFAULT_PACKAGE_GIT_URL="https://github.com/sdimitro/dlpx-aws.git"
+DEFAULT_PACKAGE_VERSION=1.0.0
```
`DEFAULT_PACKAGE_VERSION` is not needed here (as a matter of fact, I'm planning to get rid of this variable entirely in the future).

sdimitro commented:
@pzakha this should be good for another round of review.

I fixed the minor nits, updated all the repo links with the new Delphix repos, and took out any barriers that I had put for `merge_with_upstream`.

When all is good I'm squash everything in to a single commit so we can merge it.
